examples/stable_diffusion_xl/gm/modules/diffusionmodules/openaimodel.py

```diff
@@ -787,11 +789,26 @@ def construct(self, x, timesteps=None, context=None, y=None, **kwargs):
             emb = emb + self.label_emb(y)
 
         h = x
-        for module in self.input_blocks:
+        adapter_idx = 0
+        for i, module in enumerate(self.input_blocks):
             h = module(h, emb, context)
+
+            if i in [5, 8] and adapter_states:  # include adapter features in hidden states for 2nd and 3rd blocks
+                h += adapter_states[adapter_idx]
+                adapter_idx += 1
+
             hs += (h,)
+
+            if i == 3 and adapter_states:  # do not include adapter features in hidden states after the first block
+                h += adapter_states[adapter_idx]
+                adapter_idx += 1
+
             hs_idx += 1
+
         h = self.middle_block(h, emb, context)
+        if adapter_states:
+            h += adapter_states[adapter_idx]
+
         for module in self.output_blocks:
```

Comment:
Perhaps we should add a new class that inherits from UNetModel to manage the adapter. The addition of new branches may affect the original compilation process and time.

Reply:
Good suggestion. Extracted T2I-Adapter's UNet in a separate module.